Automattic/simplenote-macos
# Add spotlight support for Simplenote

### Fix
It was requested in #1000 to index all of the notes in Simplenote so they can be searched for in spotlight.  This is a great idea, so I have done just that!

With this PR you will be able to type in the text or title of any note into spotlight and the not will appear in the search results.  Then you will be able to launch Simplenote directly to that note from spotlight.

fixes #1000 

### Test
Prep:
We want to confirm indexing and removing indexing before confirming the one time indexing of all of the existing notes.  So at first we want to not run that indexing.... so:

1. Go to SimplenoteAppDelegate L.126 and comment out the line `[self indexSpotlightItemsIfNeeded];` so we don't index all your notes right away

Test adding and removing notes from the spotlight index:
1. Open up Simplenote on another device that is logged into the same account and look for a note with some text.  Open up spotlight and type that text in to the search field.  Confirm that the note does not appear
2. Open Simplenote on this device, open up that same note, add some text to it and then open a different note to force a save
3. Return to Spotlight and search for some of the text in that note.  Confirm that the note appears in spotlight as a search result
4. select that search result.  Confirm that Simplenote opens and goes to the note in question
5. Now put that note in the trash.  Go back to spotlight and search the same text, confirm that you do not see the note appear.


One time index:
6. now uncomment out the code we disabled before in the SimplenoteAppDelegate
7. relaunch the app.  Give the app some time to do the indexing, this will happen in the background so you won't notice or get any indication that it is doing so.
8. Now search for text from any of your notes.  Confirm it appears in spotlight
9. Put a break point in SimplenoteAppDelegate L602.  Run the app again and confirm you do not hit this breakpoint.  We only want to run that indexing once.

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer is required to review these changes, but anyone can perform the review.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
> `RELEASE-NOTES.txt` was updated in ca5d1a with:
> 
> > Added spotlight search support for notes
